### PR TITLE
Introduce OKX Wallet Web3 provider and authentication strategy

### PR DESCRIPTION
**Pull request by chanioxaris**
> [!IMPORTANT]
> 🔎 Previews:
> - https://clerk.com/docs/pr/1768/authentication/configuration/sign-up-sign-in-options
> - https://clerk.com/docs/pr/1768/customization/elements/reference/sign-in
> - https://clerk.com/docs/pr/1768/references/javascript/clerk/clerk
> - https://clerk.com/docs/pr/1768/references/javascript/sign-in/authenticate-with
> - https://clerk.com/docs/pr/1768/references/javascript/sign-in/first-factor
> - https://clerk.com/docs/pr/1768/references/javascript/sign-in/sign-in
> - https://clerk.com/docs/pr/1768/references/javascript/sign-up/authenticate-with
> - https://clerk.com/docs/pr/1768/references/javascript/sign-up/sign-up
> - https://clerk.com/docs/pr/1768/references/javascript/sign-up/verification
> - https://clerk.com/docs/pr/1768/references/javascript/sign-up/web3-verification
> - https://clerk.com/docs/pr/1768/references/javascript/types/sign-in-first-factor
> - https://clerk.com/docs/pr/1768/references/javascript/web3-wallet/verification
> - https://clerk.com/docs/pr/1768/references/javascript/web3-wallet/web3-wallet
> - https://clerk.com/docs/pr/1768/references/react/use-sign-in
> - https://clerk.com/docs/pr/1768/users/web3

### Explanation:

- 

### This PR:

- 

### Checklist

- [ ] I have clicked on "Files changed" and performed a thorough self-review
- [ ] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [ ] All existing checks pass

### CODE REVIEW

File: docs/references/javascript/clerk/clerk.mdx [hunk 1]
```diff
@@ -198,6 +198,39 @@ function authenticateWithCoinbaseWallet({
...
+  - `string | undefined`
+
+  Full URL or path to navigate to after a successful sign in or sign up.
+
+  ---
+
+  - `signUpContinueUrl?`
+  - `string | undefined`
+
+  The location to continue navigation to in the sign up process if data is missing.
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/references/javascript/clerk/clerk.mdx:224]: This is copy paste from the methods above. Should we update those as well?

File: docs/references/javascript/clerk/clerk.mdx [hunk 2]
```diff
@@ -198,6 +198,39 @@ function authenticateWithCoinbaseWallet({
...
+
+  ---
+
+  - `customNavigate?`
+  - `((to: string) => Promise<unknown> | unknown) | undefined`
+
+  Allows you to define a custom navigation function
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/references/javascript/clerk/clerk.mdx:231]: Same here this is copy paste from the methods above

> Is this for sign-in and sign-up?
>

Both 😃

File: docs/references/javascript/clerk/clerk.mdx [hunk 3]
```diff
@@ -198,6 +198,39 @@ function authenticateWithCoinbaseWallet({
...
+
+```typescript
+function authenticateWithOKXWallet({
+  redirectUrl,
+  signUpContinueUrl,
+  customNavigate,
+}?: AuthenticateWithOKXWalletParams): Promise<void>
+```
+
+#### `AuthenticateWithOKXWalletParams`
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/references/javascript/clerk/clerk.mdx:211]: Here it is https://github.com/clerk/javascript/blob/10c447d1a8788db825a1f8db2cefed036a167308/packages/types/src/clerk.ts#L1451

File: docs/references/javascript/sign-in/sign-in.mdx [hunk 4]
```diff
@@ -128,6 +128,7 @@ function create(params: SignInCreateParams): Promise<SignIn>
   - `phone_code`: User will receive a one-time authentication code in their phone, via SMS. At least one phone number should be on file for the user. The `phone_number_id` parameter can also be specified to select which of the user's known phone numbers the SMS will go to.
   - `web3_metamask_signature`: The verification will attempt to be completed using the user's Web3 wallet address via [Metamask](https://metamask.io/). The `web3_wallet_id` parameter can also be specified to select which of the user's known Web3 wallets will be used.
   - `web3_coinbase_wallet_signature`: The verification will attempt to be completed using the user's Web3 wallet address via [Coinbase Wallet](https://www.coinbase.com/wallet). The `web3_wallet_id` parameter can also be specified to select which of the user's known Web3 wallets will be used.
+  - `web3_okx_wallet_signature`: The verification will attempt to be completed using the user's Web3 wallet address via [OKX Wallet](https://www.okx.com/help/section/faq-web3-wallet). The `web3_wallet_id` parameter can also be specified to select which of the user's known Web3 wallets will be used.
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/references/javascript/sign-in/sign-in.mdx:131]: You can have a look here https://github.com/clerk/javascript/blob/10c447d1a8788db825a1f8db2cefed036a167308/packages/types/src/web3.ts#L13

File: docs/users/web3.mdx [hunk 5]
```diff
@@ -7,19 +7,19 @@ Learn how to use Clerk to quickly and easily add secure authentication and user
...
 
 ## Accessing the Web3 address from the frontend
 
-At this point, you should have a Next.js application integrated with Clerk and with Metamask or Coinbase Wallet authentication enabled. Run `npm run dev` and visit `localhost:3000`. If you followed the guide, all of the pages to your application should be protected, and you should see your sign-in page.
+At this point, your Next.js application should be integrated with Clerk and configured for authentication using Metamask, Coinbase Wallet, or OKX Wallet. Run `npm run dev` and visit `localhost:3000`. If you followed the guide, all pages in your application will be protected, and you should see your sign-in page.
 
-After signing in with Metamask or Coinbase Wallet, you'll be presented with the Next.js default start screen. Let's modify the start screen to display a user's primary Web3 address on the page.
+After signing in with Metamask, Coinbase Wallet, or OKX Wallet, you'll be presented with the Next.js default start screen. Let's modify the start screen to display a user's primary Web3 address on the page.
```
Comment (alexisintech) [inline docs/users/web3.mdx:22]:
```suggestion
After signing in with your Web3 provider, you'll be presented with the Next.js default start screen. Let's modify the start screen to display a user's primary Web3 address on the page.
```

File: docs/references/javascript/clerk/clerk.mdx [hunk 6]
```diff
@@ -188,14 +188,47 @@ function authenticateWithCoinbaseWallet({
...
+</Properties>
+
+### `authenticateWithOKXWallet()`
+
+```typescript
+function authenticateWithOKXWallet({
+  redirectUrl,
+  signUpContinueUrl,
+  customNavigate,
+}?: AuthenticateWithOKXWalletParams): Promise<void>
```
Comment (alexisintech) [inline docs/references/javascript/clerk/clerk.mdx:208]:
```suggestion
function authenticateWithOKXWallet: (props?: AuthenticateWithOKXWalletParams): Promise<void>
```

File: docs/references/javascript/clerk/clerk.mdx [hunk 7]
```diff
@@ -188,14 +188,47 @@ function authenticateWithCoinbaseWallet({
...
+
+  The URL to navigate to in the sign-up process if data is missing.
+
+  ---
+
+  - `customNavigate?`
+  - `((to: string) => Promise<unknown> | unknown) | undefined`
+
+  Allows you to define a custom function to handle sign-up and sign-in navigation
 </Properties>
```
Comment (alexisintech) [inline docs/references/javascript/clerk/clerk.mdx:232]: looks like `unsafeMetadata` and `legalAccepted` are missing

Screenshot 2024-12-06 at 14 03 48

File: docs/references/javascript/clerk/clerk.mdx [hunk 8]
```diff
@@ -181,21 +181,54 @@ function authenticateWithCoinbaseWallet({
...
 
   - `customNavigate?`
   - `((to: string) => Promise<unknown> | unknown) | undefined`
 
-  Allows you to define a custom navigation function
+  A function that overrides Clerk's default navigation behavior, allowing custom handling of navigation during sign-up and sign-in flows.
+</Properties>
+
+### `authenticateWithOKXWallet()`
+
```
Comment (alexisintech) [inline docs/references/javascript/clerk/clerk.mdx:202]:
```suggestion
Starts a sign-in flow that uses the OKX Wallet to authenticate the user using their Web3 wallet address.

```

File: docs/references/javascript/sign-in/authenticate-with.mdx [hunk 9]
```diff
@@ -66,6 +66,20 @@ function authenticateWithCoinbaseWallet(): Promise<SignInResource>
...
+```typescript
+function authenticateWithOKXWallet(): Promise<SignInResource>
+```
+
+#### Returns
+
+| Type | Description |
+| - | - |
+| `Promise<SignInResource>` | A `Promise` which resolves to the current [`SignIn`][signin-ref]. |
+
```
Comment (alexisintech) [inline docs/references/javascript/sign-in/authenticate-with.mdx:82]: you didn't know this, so it's okay, but we want to remove the "#### Returns" sections from the docs. I'm thinking we can rephrase like this. This way, the user can find the resource's reference (the link to the signin object) if they want it.

````suggestion
Starts a sign-in flow that uses the OKX Wallet to authenticate the user using their Web3 wallet address. Returns `Promise` which resolves to the current [`SignIn`][signin-ref].

```typescript
function authenticateWithOKXWallet(): Promise<SignInResource>
```
````

File: docs/references/javascript/sign-in/first-factor.mdx [hunk 10]
```diff
@@ -49,7 +50,7 @@ function prepareFirstFactor(params: PrepareFirstFactorParams): Promise<SignIn>
   - `web3WalletId?`
   - `string`
 
-  Unique identifier for the user's Web3 wallet address. This parameter will work only when the `web3_metamask_signature` or `web3_coinbase_wallet_signature` strategy is specified.
+  Unique identifier for the user's Web3 wallet address. This parameter will work only when the `web3_metamask_signature`, `web3_coinbase_wallet_signature` or `web3_okx_wallet_signature` strategy is specified.
```
Comment (alexisintech) [inline docs/references/javascript/sign-in/first-factor.mdx:53]:
```suggestion
  Unique identifier for the user's Web3 wallet address. This parameter will work only when the `strategy` is set to `web3_metamask_signature`, `web3_coinbase_wallet_signature`, or `web3_okx_wallet_signature`.
```

File: docs/references/react/use-sign-in.mdx [hunk 11]
```diff
@@ -77,7 +77,7 @@ The `status` property of the `SignIn` object can be one of the following values:
 | Values | Descriptiom |
 | - | - |
 | `complete` | The user has been signed in and custom flow can proceed to `setActive()` to create session. |
-| `needs_first_factor` | The First Factor verification is missing. One of `email_link`, `email_code`, `phone_code`, `web3_metamask_signature`, `web3_coinbase_wallet_signature` or `oauth_provider` is required to verify user. See [First Factor](/docs/references/javascript/sign-in/first-factor) for details. |
+| `needs_first_factor` | The First Factor verification is missing. One of `email_link`, `email_code`, `phone_code`, `web3_metamask_signature`, `web3_coinbase_wallet_signature`, `web3_okx_wallet_signature` or `oauth_provider` is required to verify user. See [First Factor](/docs/references/javascript/sign-in/first-factor) for details. |
```
Comment (alexisintech) [inline docs/references/react/use-sign-in.mdx:80]:
```suggestion
| `needs_first_factor` | The first factor verification is missing. One of `email_link`, `email_code`, `phone_code`, `web3_metamask_signature`, `web3_coinbase_wallet_signature`, `web3_okx_wallet_signature` or `oauth_provider` is required to verify the user. See [the reference](/docs/references/javascript/sign-in/first-factor) for more information. |
```

File: docs/references/javascript/clerk/clerk.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexisintech) [reply docs/references/javascript/clerk/clerk.mdx]: can it be a path? or it requires the full URL? @chanioxaris

File: docs/references/javascript/clerk/clerk.mdx (hunk 7 above)
Comment (alexisintech) [reply docs/references/javascript/clerk/clerk.mdx:244]: handled here for ya [4f8bdb8](https://github.com/clerk/clerk-docs/pull/1768/commits/4f8bdb810f2309f05eb3162697481189df4c595d)

File: docs/references/javascript/clerk/clerk.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/references/javascript/clerk/clerk.mdx]: Yes it can also be a path

**Review by alexisintech — APPROVED** [review]
looks ready to merge when you're ready! 😸💖

**Pull request merged**